test/performance/range/views/view_translate_benchmark.cpp

```diff
@@ -0,0 +1,209 @@
+
+    if constexpr (std::is_same_v<tag_t, baseline_tag>)
+    {
+        std::vector<std::vector<aa27>> cpy = c | views::translate_join | views::to<std::vector<std::vector<aa27>>>;
+        sequential_read_impl(state, cpy);
+    }
+    else if constexpr (std::is_same_v<tag_t, translate_tag>)
+    {
+        auto v = c | views::translate | views::join;
+        sequential_read_impl(state, v);
+    }
+    else
+    {
+        auto v = c | views::translate_join;
+        sequential_read_impl(state, v);
+    }
+}
+
+BENCHMARK_TEMPLATE(sequential_read, baseline_tag);
+BENCHMARK_TEMPLATE(sequential_read, translate_tag);
+BENCHMARK_TEMPLATE(sequential_read, translate_join_tag);
+
+// ============================================================================
+//  random_access
+// ============================================================================
+
+template <typename rng_t>
+void random_access_impl(benchmark::State & state, rng_t && rng, std::vector<size_t> const & access_positions)
+{
+    for (auto _ : state)
+    {
+        for (auto pos : access_positions)
+        {
+            benchmark::DoNotOptimize(rng[pos][0].to_rank());
+        }
+    }
+}
+
+template <typename tag_t>
+void random_access(benchmark::State & state)
+{
+    std::vector<std::vector<dna4>> c;
+    c.resize(1000);
+
+    for (size_t i = 0; i < c.size(); ++i)
+        c[i] = test::generate_sequence<seqan3::dna4>(100, 0, 0);
+
+    std::vector<size_t> access_positions;
+    access_positions.resize(200);
+    std::mt19937 gen(42);
+    std::uniform_int_distribution<size_t> dis_position(0, 1000 - 1);
+
+    for (size_t i = 0; i < 200; ++i)
+        access_positions[i] = dis_position(gen);
+
+    if constexpr (std::is_same_v<tag_t, baseline_tag>)
+    {
+        std::vector<std::vector<aa27>> cpy = c | views::translate_join | views::to<std::vector<std::vector<aa27>>>;
+        random_access_impl(state, cpy, access_positions);
+    }
+    else
+    {
+        auto v = c | views::translate_join;
+        random_access_impl(state, v, access_positions);
+    }
+}
+
+BENCHMARK_TEMPLATE(random_access, baseline_tag);
+BENCHMARK_TEMPLATE(random_access, translate_join_tag);
+
+// ============================================================================
+//  copy_vector
+// ============================================================================
+
+template <typename adaptor_t>
+void copy_impl(benchmark::State & state, std::vector<std::vector<dna4>> const & c, adaptor_t & adaptor)
+{
+    for (auto _ : state)
+    {
+        std::vector<std::vector<aa27>> cpy;
+        benchmark::DoNotOptimize(cpy = c | adaptor | views::to<std::vector<std::vector<aa27>>>);
+    }
+}
+
+#ifdef SEQAN3_HAS_SEQAN2
+template <typename tag_t>
+void copy_impl_seqan2(benchmark::State & state, seqan::StringSet<seqan::DnaString> const & c)
+{
+    for (auto _ : state)
+    {
+        seqan::StringSet<seqan::String<seqan::AminoAcid>, seqan::Owner<seqan::ConcatDirect<> > > out;
```

Comment:
The version that accepts `Owner<ConcatDirect<>>` is particularly optimised. Maybe you could add another one that just takes `Owner<>`? Could just be another template parameter...